### PR TITLE
DDF-6342 Remove unused Kernel features

### PR DESCRIPTION
**Pull request by SmithJosh**
#### What does this PR do?
Removes unused features from the kernel. This shrinks the kernel from 120MB to 89MB.
This also resolves some CVEs as the features were pulling in some old dependencies with known vulnerabilities. 

#### Who is reviewing it? 
@blen-desta @emmberk @garrettfreibott @stustison 

#### Select relevant component teams: 
@codice/security 

#### How should this be tested?
CI

#### Any background context you want to provide?
The fabric8 features repo was originally added to provide health and readiness endpoints for DDF: https://github.com/codice/ddf/pull/3155
But later the features were turned off by default. Apparently there were issues with system startup when they were turned on: https://github.com/codice/ddf/pull/3302

#### What are the relevant tickets?
Fixes: #6342

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by SmithJosh** [comment]
build now

**Review by jlcsmith — APPROVED** [review]

**Comment by SmithJosh** [comment]
build now

**Comment by SmithJosh** [comment]
build now

**Review by emmberk — APPROVED** [review]

**Review by garrettfreibott — APPROVED** [review]

**Review by stustison — APPROVED** [review]

File: features/kernel/src/main/feature/feature.xml [hunk 1]
```diff
@@ -17,8 +17,6 @@
           xsi:schemaLocation="http://karaf.apache.org/xmlns/features/v1.3.0 http://karaf.apache.org/xmlns/features/v1.3.0">
 
     <repository>mvn:org.apache.karaf.features/standard/${karaf.version}/xml/features</repository>
-    <repository>mvn:org.apache.karaf.features/spring/${karaf.version}/xml/features</repository>
-    <repository>mvn:io.fabric8/fabric8-karaf-features/${fabric8.version}/xml/features</repository>
```
Comment (emmberk) [inline features/kernel/src/main/feature/feature.xml:21]: ✏️ I don't see `fabric8.version` being used anywhere else. Can that property be removed in the root pom?

Comment (SmithJosh) [reply features/kernel/src/main/feature/feature.xml:21]: Yep, I removed it. Thanks

**Comment by SmithJosh** [comment]
build now

**Comment by SmithJosh** [comment]
build now

**Comment by SmithJosh** [comment]
Cleaned up some additional changes that were made when the fabric8 features were introduced.

build now

**Comment by SmithJosh** [comment]
Closing this in favor of https://github.com/codice/ddf/pull/6370. The kernel was only part of the problem. There were a lot more unused features in the main distro.

**Pull request closed without merging**